truss-chains/truss_chains/deploy.py

```diff
@@ -183,6 +183,20 @@ def deploy_remotely(
         framework.global_chainlet_registry.get_descriptor(entrypoint).name,
         name=options.chain_name,
     )
+    if isinstance(options, definitions.DeploymentOptionsBaseten):
+        secrets_info = options.remote_provider.api.get_all_secrets()
+        secret_names = {sec["name"] for sec in secrets_info["secrets"]}
+        if definitions.BASETEN_API_SECRET_NAME not in secret_names:
+            logging.info(
+                "It seems you are using chains for the first time, since there "
+                f"is no `{definitions.BASETEN_API_SECRET_NAME}` secret on baseten. "
+                "Will create this secret automatically."
+            )
+            options.remote_provider.api.upsert_secret(
```

Comment:
how should error handling work here? Could we show an error message like 'Could not contact server' in the case of an HTTP error

Reply:
Hm, I don't see a reason why not to rely on the already existing `resp.raise_for_status()` in the API implementation - that will raise an exception with the most appropriate message/error code.